adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala

```diff
@@ -339,6 +339,373 @@ private[adam] class VariantContextConverter extends Serializable with Logging {
+  )
+
+  private[converters] def extractAncestralAllele(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getAncestralAllele).foreach(vcb.attribute("AA", _))
+    vcb
+  }
+
+  private[converters] def extractDbSnp(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getDbSnp).foreach(vcb.attribute("DB", _))
+    vcb
+  }
+
+  private[converters] def extractHapMap2(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getHapMap2).foreach(vcb.attribute("H2", _))
+    vcb
+  }
+
+  private[converters] def extractHapMap3(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getHapMap3).foreach(vcb.attribute("H3", _))
+    vcb
+  }
+
+  private[converters] def extractThousandGenomes(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getThousandGenomes).foreach(vcb.attribute("1000G", _))
+    vcb
+  }
+
+  private[converters] def extractCigar(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getCigar).foreach(vcb.attribute("CIGAR", _))
+    vcb
+  }
+
+  private[converters] def extractAlleleCount(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getAlleleCount).foreach(vcb.attribute("AC", _))
+    vcb
+  }
+
+  private[converters] def extractAlleleFrequency(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getAlleleCount).foreach(vcb.attribute("AF", _))
+    vcb
+  }
+
+  private[converters] def extractReadDepth(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getReadDepth).foreach(rd => vcb.attribute("AD", prependDefaultReferenceValue(rd)))
+    vcb
+  }
+
+  private[converters] def extractForwardReadDepth(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getForwardReadDepth).foreach(frd => vcb.attribute("ADF", prependDefaultReferenceValue(frd)))
+    vcb
+  }
+
+  private[converters] def extractReverseReadDepth(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getReverseReadDepth).foreach(rrd => vcb.attribute("ADR", prependDefaultReferenceValue(rrd)))
+    vcb
+  }
+
+  private[converters] def extractTranscriptEffects(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    if (!va.getTranscriptEffects.isEmpty) {
+      vcb.attribute("ANN", TranscriptEffectConverter.convertToVcfInfoAnnValue(va.getTranscriptEffects))
+    }
+    vcb
+  }
+
+  // reference value is missing for Number=R VCF INFO attributes
+  private[converters] def prependDefaultReferenceValue(v: Int): String = {
+    "-1," + v
```

Comment:
Alternatively, you could just not prepend these, and ignore index = 0 for Number=R fields.